Libraries/PeriphDrivers/Include/MAX78002/spi.h

```diff
@@ -1039,6 +1028,48 @@ int MXC_SPI_SlaveTransactionAsync(mxc_spi_req_t *req);
  */
 int MXC_SPI_SlaveTransactionDMA(mxc_spi_req_t *req);
 
+/**
+ * @brief   Setup an interrupt-driven, non-blocking SPI Target transaction.
+ * 
+ * @param   spi         Pointer to SPI instance's registers.
+ * @param   tx_buffer   Pointer to transmit buffer (in terms of bytes).
+ * @param   tx_fr_len   Number of frames to transmit from transmit buffer.
+ * @param   rx_buffer   Pointer to transmit buffer (in terms of bytes).
+ * @param   rx_fr_len   Number of frames to store in recieve buffer.
+ *
+ * @return  See \ref MXC_Error_Codes for the list of error return codes.
+ */
+int MXC_SPI_TargetTransaction(mxc_spi_regs_t *spi, uint8_t *tx_buffer, uint32_t tx_fr_len,
```

Comment:
We're changing the arguments from a request struct to a list of parameters.
1. This voids `TargetTransaction` as a drop-in replacement for `MasterTransaction`.
2. We will have problems in the future if/when we need to modify these arguments.  The previous struct gives us that flexibility.

Reply:
1. Are you referring to SPI_SlaveTransaction instead? (Master => Controller :: Slave => Target). Note, the SlaveTransaction function still exists.
2. I'm doing this as a continuation from the ControllerTransaction Functions when it was decided to use this list of input parameters instead of the request struct.

Reply:
Are we electing to use request structs for the transactions now? The ControllerTransaction functions don't use the request struct because it was previously decided that the list of parameters was preferred for the transaction functions, but not the init function.

Reply:
I'm getting confused about what we're trying to preserve across v1 and v2...  So the Controller/Target functions are not intended as drop-in replacements for Master/Slave?  They're just for v2?

Reply:
The SPI v2 Controller/Target transaction functions weren't intended as a drop-in replacement. It was technically supposed to be a new SPI library that still supported the v1 prototypes because you preferred using a list of parameters in your fastspi library, so that's what I was referencing. However, I've updated the transaction functions to pass in request structs instead.